File: objectdetection/src/main/java/com/andruid/magic/objectdetection/Recognition.java
package com.andruid.magic.objectdetection;

import android.graphics.RectF;

public class Recognition {
    private final int id;
    private final String title;
    private final Float confidence;
    private RectF location;

    Recognition(final int id, final String title, final Float confidence, final RectF location) {
        this.id = id;
        this.title = title;
        this.confidence = confidence;
        this.location = location;
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return title;
    }

    public Float getConfidence() {
        return confidence;
    }

    public RectF getLocation() {
        return new RectF(location);
    }

    public void setLocation(RectF location) {
        this.location = location;
    }
}

File: objectdetection/src/main/java/com/andruid/magic/objectdetection/model/Recognition.kt
package com.andruid.magic.objectdetection.model

import android.graphics.RectF

/**
 * Model class for all results returned from the object detection model
 * @property id recognition result id
 * @property title label of the object
 * @property confidence confidence returned by the model
 * @property location rectangular coordinates of object in the image
 */
data class Recognition(
        val id: Int,
        val title: String,
        val confidence: Float,
        val location: RectF
)